llext: Use clang format on all llext files

teburd opened the pull request:
If we stick with clang-format we don't need to nitpick over little spacing issues for new contributions, we can simply ask people to run the formatting tool. Much simpler.

Noticed these files weren't formatted with the tool in #83423 and the result is nits over spacing which are not the kind of review comments I want to see, use the tool to ensure spacing/formatting nits are unnecessary.

Commit:
llext: Use clang format on all llext files

If we stick with clang-format we don't need to nitpick over little
spacing issues for new contributions, we can simply ask people to run
the formatting tool. Much simpler.

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

teburd commented on subsys/llext/llext_load.c:
```diff
@@ -594,8 +584,8 @@ static int llext_export_symbols(struct llext_loader *ldr, struct llext *ext,
-		exp_tab->sym_cnt = ldr->sects[LLEXT_MEM_EXPORT].sh_size
-				   / sizeof(struct llext_symbol);
+		exp_tab->sym_cnt =
+			ldr->sects[LLEXT_MEM_EXPORT].sh_size / sizeof(struct llext_symbol);
```
The project has a clang format config file, I’ll see if there’s an option for this

pillo79 reviewed (commented):
`clang-format` is a great tool to enforce a minimum standard and most of the changes here are great, but personally I'm against requiring absolute compliance: I have highlighted several places where word wraps were chosen to maximize immediate pattern recognition; with the "cleanup" these are now more difficult to spot.

So soft -1 (to the rule), but ofc I’m willing to go with whatever is ultimately decided.

Commit:
clang-format: Adjust PenaltyBreak* values

These options tell clang-format penalize the formatter for trying to
break a long line up on particular elements like assignment, opening
brackets, and such. Adjusting these means long lines are broken at more
natural places that match existing code.

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

teburd commented:
Ultimately most of the format changes here are quite nice to me. It's a thoughtless excercise to run clang-format and avoids formatting nits which I really never like seeing (I want to see quality structural review comments).

But if the tool can't be made, with its many options, to get us like 95% success with a few whiffs (the tool can always be disabled for small sections like logically aligned printfs) then its just not good enough still which is disappointing. I'll tinker with the rules a bit more today to try and address the comments from @pillo79 and @lyakh (I appreciate you two looking this over!).
